### PR TITLE
feat(async-queue): add async-queue package

### PR DESCRIPTION
**Pull request by Favna**

### CODE REVIEW

File: packages/async-queue/src/lib/AsyncQueue.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	public wait(): Promise<void> {
+		const next = this.promises.length ? this.promises[this.promises.length - 1].promise : Promise.resolve();
+		// eslint-disable-next-line @typescript-eslint/init-declarations
+		let resolve: () => void;
+		const promise = new Promise<void>((res) => {
+			resolve = res;
+		});
+
+		this.promises.push({
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
Comment (vladfrangu) [inline packages/async-queue/src/lib/AsyncQueue.ts:45]: is this still needed?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/async-queue/src/lib/AsyncQueue.ts (hunk 1 above)
Comment (Favna) [reply packages/async-queue/src/lib/AsyncQueue.ts:45]: oh fuck i didnt check lmfao

Comment (Favna) [reply packages/async-queue/src/lib/AsyncQueue.ts]: fixed in 68e353d

File: packages/async-queue/src/lib/AsyncQueue.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	public wait(): Promise<void> {
+		const next = this.promises.length ? this.promises[this.promises.length - 1].promise : Promise.resolve();
+		// eslint-disable-next-line @typescript-eslint/init-declarations
+		let resolve: () => void;
+		const promise = new Promise<void>((res) => {
+			resolve = res;
+		});
+
+		this.promises.push({
+			resolve: resolve!,
```
Comment (vladfrangu) [inline packages/async-queue/src/lib/AsyncQueue.ts:45]: ok but can someone explain to me why we need `!` in resolve?

File: packages/async-queue/src/lib/AsyncQueue.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	 *     }
+	 * }
+	 *
+	 * request(someUrl1, someOptions1); // Will call fetch() immediately
+	 * request(someUrl2, someOptions2); // Will call fetch() after the first finished
+	 * request(someUrl3, someOptions3); // Will call fetch() after the second finished
+	 */
+	public wait(): Promise<void> {
+		const next = this.promises.length ? this.promises[this.promises.length - 1].promise : Promise.resolve();
+		// eslint-disable-next-line @typescript-eslint/init-declarations
```
Comment (vladfrangu) [inline packages/async-queue/src/lib/AsyncQueue.ts:38]: haven't we removed this rule?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/async-queue/src/lib/AsyncQueue.ts (hunk 2 above)
Comment (Favna) [reply packages/async-queue/src/lib/AsyncQueue.ts:45]: image

File: packages/async-queue/src/lib/AsyncQueue.ts (hunk 3 above)
Comment (Favna) [reply packages/async-queue/src/lib/AsyncQueue.ts:38]: We haven't

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**